src/grblogtools/single_log_parser.py

```diff
@@ -1,42 +1,71 @@
+from grblogtools.continuous_parser import ContinuousParser
 from grblogtools.header_parser import HeaderParser
 from grblogtools.nodelog import NodeLogParser
 from grblogtools.norel import NoRelParser
 from grblogtools.presolve_parser import PresolveParser
+from grblogtools.termination import TerminationParser
 
 
 class SingleLogParser:
+    """Parses one log. The class expects start_parsing to be called once for each
+    line, until it returns true, after which it expects continue_parsing to be
+    called once for each remaining line in sequence."""
+
     def __init__(self):
 
         # Parsers in sequence.
         self.header_parser = HeaderParser()
         self.presolve_parser = PresolveParser()
         self.norel_parser = NoRelParser()
-        # Add relaxation/continuous model parser here
+        self.continuous_parser = ContinuousParser()
         self.nodelog_parser = NodeLogParser()
+        self.termination_parser = TerminationParser()
 
         # State
         self.current_parser = self.header_parser
         self.future_parsers = [
             self.presolve_parser,
             self.norel_parser,
+            self.continuous_parser,
             self.nodelog_parser,
+            self.termination_parser,
```

Comment:
One other idea to address the termination stage is to treat this parser differently because the termination messages might be scattered throughout the log. 

The idea would be that we do not include the termination parser in the list of future parsers. In the method `continue_parser()` of the `SingleLogParser()`, if the line is not already matched with a pattern in the current parser and if the line does not match the start pattern of any other parser, we pass the line to the termination parser (we just need to have one method named `parse` for this parser) and save the information in a dict (the same as we do in the other parsers). 

This will work because the lines that appear later in the log file will always overwrite whatever optimal objective value and status we have already saved in the termination parser.

Reply:
Sounds good! I'll add an issue describing the change and assign it to you.